symbol-table/scope-element.dart

```diff
@@ -6,6 +6,8 @@ import '../ast-nodes/index.dart';
 /// An abstract element in the scope's linked list (scope chain).
 abstract class ScopeElement {
   ScopeElement next;
+  
+  bool readOnly = false;
```

Comment:
This should only be on scope declarations

Reply:
But why, if we everywhere have ScopeElement scopeMark? I mean it would work, but code will be a little bit uglier.

Reply:
Not uglier, it would be more correct. Whenever we talk about `readOnly`, we must be talking about declarations, therefore, whenever we want to refer to that field, we should make an explicit cast of `ScopeElement` to `ScopeDeclaration`. If we've made a mistake somewhere and had another scope element where we expected a scope declaration, this cast will fail at runtime and we will be aware that we've made a mistake